modules/core/src/main/scala/gem/config/Gmos.scala

```diff
@@ -4,19 +4,108 @@
+    * This class essentially provides a newtype for Int.
+    */
+  sealed abstract case class GmosShuffleOffset(detectorRows: Int)
+
+  object GmosShuffleOffset {
+
+    /** Constructs the shuffle offset with the given number of detector rows,
+      * provided it is a positive number.
+      *
+      * @return `Some(GmosShuffleOffset(rows))` if `rows` is positive,
+      *         `None` otherwise
+      */
+    def fromRowCount(rows: Int): Option[GmosShuffleOffset] =
+      (rows > 0) option new GmosShuffleOffset(rows) {}
+
+    /** Constructs the shuffle offset with the given number of detector rows
+      * provided `rows` is positive, or throws an exception if zero or negative.
+      */
+    def unsafeFromRowCount(rows: Int): GmosShuffleOffset =
+      fromRowCount(rows).getOrElse(sys.error(s"Expecting positive detector row count, not $rows"))
+
+    /** Constructs a shuffle offset using the default number of detector rows
+      * associated with the detector.
+      */
+    def defaultFromDetector(detector: GmosDetector): GmosShuffleOffset =
+      fromRowCount(detector.shuffleOffset).getOrElse(sys.error(s"Misconfigured GmosDetector $detector"))
+
+    implicit val EqualGmosShuffleOffset: Equal[GmosShuffleOffset] =
+      Equal.equalA
+  }
+
+  /** The number of nod-and-shuffle cycles, which must be at least 1. This class
+    * essentially provides a newtype for Int.
+    */
+  sealed abstract case class GmosShuffleCycles(toInt: Int)
+
+  object GmosShuffleCycles {
+
+    /** Default non-and-shuffle cycles, which is 1. */
+    val Default: GmosShuffleCycles =
+      unsafeFromCycleCount(1)
+
+    /** Constructs the shuffle cycles from a count if `cycles` is positive.
+      *
+      * @return `Some(GmosShuffleCycles(cycles))` if `cycles` is positive,
+      *         `None` otherwise
+      */
+    def fromCycleCount(cycles: Int): Option[GmosShuffleCycles] =
+      (cycles > 0) option new GmosShuffleCycles(cycles) {}
+
+    /** Constructs the shuffle cycles with the given `cycles` count provided it
+      * is positive, or else throws an exception if 0 or negative.
+      */
+    def unsafeFromCycleCount(cycles: Int): GmosShuffleCycles =
+      fromCycleCount(cycles).getOrElse(sys.error(s"Expecting positive shuffle cycles, not $cycles"))
+
+    implicit val EqualGmosShuffleCycles: Equal[GmosShuffleCycles] =
+      Equal.equalA
+  }
+
+  // TODO: there are many ways to misconfigure Nod And Shuffle.  Some of these
+  // ways can be caught in a companion object constructor, and some cannot, or
+  // at least cannot easily.  In the first category, we should definitly check
+  // whether posA and posB are close enough together (< 2 arcsecs) to allow
+  // e-offsetting.  In the second category, shuffle offset in detector rows has
+  // to be a multiple of y-binning.  Worse, y-binning is part of the dynamic
+  // configuration so it isn't clear what happens if the shuffle offset isn't
+  // always a multiple of y-binning.
```

Comment:
This seems like the kind of consistency check we will need to run after the fact, and report it as user error.